File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/java/io/github/lishangbu/avalon/dataset/service/MoveDamageClassService.java
package io.github.lishangbu.avalon.dataset.service;

import io.github.lishangbu.avalon.dataset.entity.MoveDamageClass;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/// 招式伤害类别服务。
public interface MoveDamageClassService {

    /// 根据条件分页查询招式伤害类别。
    Page<MoveDamageClass> getPageByCondition(MoveDamageClass moveDamageClass, Pageable pageable);

    /// 新增招式伤害类别。
    MoveDamageClass save(MoveDamageClass moveDamageClass);

    /// 更新招式伤害类别。
    MoveDamageClass update(MoveDamageClass moveDamageClass);

    /// 根据主键删除招式伤害类别。
    void removeById(Long id);

    /// 根据条件查询招式伤害类别列表。
    List<MoveDamageClass> listByCondition(MoveDamageClass moveDamageClass);
}

File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/kotlin/io/github/lishangbu/avalon/dataset/service/MoveDamageClassService.kt
package io.github.lishangbu.avalon.dataset.service

import io.github.lishangbu.avalon.dataset.entity.dto.MoveDamageClassSpecification
import io.github.lishangbu.avalon.dataset.entity.dto.MoveDamageClassView
import io.github.lishangbu.avalon.dataset.entity.dto.SaveMoveDamageClassInput
import io.github.lishangbu.avalon.dataset.entity.dto.UpdateMoveDamageClassInput
import org.babyfish.jimmer.Page
import org.springframework.data.domain.Pageable

/** 招式伤害分类服务 */
interface MoveDamageClassService {
    /** 按条件分页查询招式伤害分类*/
    fun getPageByCondition(
        specification: MoveDamageClassSpecification,
        pageable: Pageable,
    ): Page<MoveDamageClassView>

    /** 保存招式伤害分类 */
    fun save(command: SaveMoveDamageClassInput): MoveDamageClassView

    /** 更新招式伤害分类 */
    fun update(command: UpdateMoveDamageClassInput): MoveDamageClassView

    /** 按 ID 删除招式伤害分类 */
    fun removeById(id: Long)

    /** 根据条件查询招式伤害分类列表 */
    fun listByCondition(specification: MoveDamageClassSpecification): List<MoveDamageClassView>
}
